zerver/lib/actions.py

```diff
@@ -583,6 +583,7 @@ def notify_created_user(user_profile: UserProfile) -> None:
         # later event.
         custom_profile_field_data={},
     )
+    person["email_address_visibility"] = user_row["email_address_visibility"]
```

Comment:
Sending `email_address_visibility` here to compute `client_gravata`r in apply_event. It is removed in the user objects passed to clients as response of different endpoints.

Reply:
OK, that's reasonable.

Reply:
Actually, I'm not sure why `apply_event` would need access to this -- shouldn't the user get an event with `delivery_email` present or not depending on their role?

Reply:
We set `client_gravatar` to false only depending upon the `email_address_visibility` of user, irrespective of role. You also left a comment before about we cannot plumb the user role as of now.